lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php

```diff
@@ -372,4 +422,45 @@ private function castObject(stdClass $sourceObject, $destinationClass, array $ct
 
         return $destinationClass;
     }
+
+    /**
+     * @return resource
+     *
+     * @throws DB2Exception
+     */
+    private function createTemporaryFile()
+    {
+        $handle = @tmpfile();
+
+        if ($handle === false) {
+            throw new DB2Exception('Could not create temporary file: ' . error_get_last()['message']);
+        }
+
+        return $handle;
+    }
+
+    /**
+     * @param resource $source
+     * @param resource $target
+     *
+     * @throws DB2Exception
+     */
+    private function copyStreamToStream($source, $target) : void
+    {
+        if (@stream_copy_to_stream($source, $target) === false) {
+            throw new DB2Exception('Could not copy source stream to temporary file: ' . error_get_last()['message']);
```

Comment:
Same as above. We could emulate the failure with a mock if streams were objects. Otherwise, we'd have to define an object wrapper for a temporary file and mock it. Too much of a trouble for an implementation detail IMO.